src/CodeGen_X86.cpp

```diff
@@ -111,29 +111,37 @@ struct x86Intrinsic {
 
 // clang-format off
 const x86Intrinsic intrinsic_defs[] = {
-    {"abs_i8x32", UInt(8, 32), "abs", {Int(8, 32)}, Target::AVX2},
-    {"abs_i16x16", UInt(16, 16), "abs", {Int(16, 16)}, Target::AVX2},
-    {"abs_i32x8", UInt(32, 8), "abs", {Int(32, 8)}, Target::AVX2},
+    {"llvm.x86.avx2.pabs.b", UInt(8, 32), "abs", {Int(8, 32)}, Target::AVX2},
+    {"llvm.x86.avx2.pabs.d", UInt(16, 16), "abs", {Int(16, 16)}, Target::AVX2},
+    {"llvm.x86.avx2.pabs.w", UInt(32, 8), "abs", {Int(32, 8)}, Target::AVX2},
     {"abs_f32x8", Float(32, 8), "abs", {Float(32, 8)}, Target::AVX2},
-    {"abs_i8x16", UInt(8, 16), "abs", {Int(8, 16)}, Target::SSE41},
-    {"abs_i16x8", UInt(16, 8), "abs", {Int(16, 8)}, Target::SSE41},
-    {"abs_i32x4", UInt(32, 4), "abs", {Int(32, 4)}, Target::SSE41},
+    {"llvm.x86.ssse3.pabs.b.128", UInt(8, 16), "abs", {Int(8, 16)}, Target::SSE41},
+    {"llvm.x86.ssse3.pabs.d.128", UInt(16, 8), "abs", {Int(16, 8)}, Target::SSE41},
+    {"llvm.x86.ssse3.pabs.w.128", UInt(32, 4), "abs", {Int(32, 4)}, Target::SSE41},
     {"abs_f32x4", Float(32, 4), "abs", {Float(32, 4)}},
 
     {"round_f32x4", Float(32, 4), "round", {Float(32, 4)}, Target::SSE41},
     {"round_f64x2", Float(64, 2), "round", {Float(64, 2)}, Target::SSE41},
     {"round_f32x8", Float(32, 8), "round", {Float(32, 8)}, Target::AVX},
     {"round_f64x4", Float(64, 4), "round", {Float(64, 4)}, Target::AVX},
 
+    {"llvm.sadd.sat.v64i8", Int(8, 64), "saturating_add", {Int(8, 64), Int(8, 64)}, Target::AVX512_Cannonlake},
+    {"llvm.sadd.sat.v64i8", Int(8, 64), "saturating_add", {Int(8, 64), Int(8, 64)}, Target::AVX512_Skylake},
```

Comment:
These look identical except for the feature flags -- what's up with that? Is it just that we want to enable it on either of these features? (other similar ones down below)

Reply:
Yes - every arch that implements AVX512BW has these instructions, which afaict is only Skylake and Cannonlake.

Reply:
I guess we don't support using a mask of features there? I guess since we support > 64 features we'd need to use a bitset or similar. Bleah. I guess this is fine since it's not common, just feels a little weird. Maybe add a comment.

Reply:
It might be worth added an AVX512BW feature flag, that gets added for Cannonlake/Skylake. I don't know what it normally takes to justify adding new feature flags.

Reply:
Oops, I think that was a timing conflict, my message wasn't in reply to your follow-up, I didn't see that. Yes, I'll add a comment. And yes, I wish it could be a mask of features.

Reply:
> AVX512BW

There was a debate about this a while back (ie: should AVX512 be based around capabilities or variants). If you wish to try to argue the case again, let me know :-)

Reply:
It would make instruction selection simpler :') But I won't argue too hard. I also just realized that Skylake appears to be a strict subset of Cannonlake (based on `complete_x86_target`), so I'm just gonna remove the Cannonlake extras.

Reply:
fixed in [f6eb8aa](https://github.com/halide/Halide/pull/7129/commits/f6eb8aad4d6fbfebf2c077bd7cab71677dff8346)